lib/delayed/backend/active_record.rb

```diff
@@ -83,8 +83,7 @@ def self.reserve_with_scope(ready_scope, worker, now)
 
         def self.reserve_with_scope_using_default_sql(ready_scope, worker, now)
           # This is our old fashion, tried and true, but slower lookup
-          bad_sql = ready_scope.select("id").limit(worker.read_ahead).to_sql
-          joins("JOIN (#{bad_sql}) ids ON ids.id = delayed_jobs.id").readonly(false).detect do |job|
+          ready_scope.limit(worker.read_ahead).detect do |job|
```

Comment:
@schiza right, forgot to mention that in description.

@karol-nowak added this because our way of querying jobs couldn't use index and went to tables. It was heavy when we had to load all records cause the handlers are large. So it was better to just load ids and then load only the full rows that we want. 

But if after the change we actually can properly use index then this thing is redundant.